### PR TITLE
Use JsVal instead of JsRef (nr. 7)

### PR DESCRIPTION
**Pull request by hoodmane**
Makes python2js do what python2js_val did and remove python2js_val.

- [x] Add a test
- [x] Update changelog

### CODE REVIEW

File: src/tests/test_typeconversions.py [hunk 1]
```diff
@@ -1576,6 +1576,17 @@ def test_dict_converter_cache(selenium):
     )
 
 
+@pytest.mark.xfail(reason="TODO: Fix me")
```
Comment (hoodmane) [inline src/tests/test_typeconversions.py:1579]: This isn't a regression, this apparently never worked correctly.

**Review by ryanking13 — APPROVED** [review]
Thanks @hoodmane!

File: src/core/jsproxy.c [hunk 2]
```diff
@@ -389,7 +389,9 @@ JsProxy_GetAttr(PyObject* self, PyObject* attr)
 
   jsresult = JsProxy_GetAttr_js(JsProxy_VAL(self), key);
   if (JsvNull_Check(jsresult)) {
-    PyErr_SetString(PyExc_AttributeError, key);
+    if (!PyErr_Occurred()) {
```
[the remark this answers is not in the document]
Comment (hoodmane) [reply src/core/jsproxy.c:392]: Yeah but a very obscure one.

[the remark this answers is not in the document]
Comment (hoodmane) [reply src/core/jsproxy.c:392]: I should add a test case I suppose.

**Pull request merged**